### PR TITLE
[WIP] Support for custom input filesystems

### PR DESCRIPTION
**Pull request by LukeSheard**
This is an extremely long overdue initial PR for #649 with what I hope is some fixes for work we've done internally to support virtual filesystems for webpack through the inputfilesystem property. I'm still sketching out the work to put this back into ts-loader, but essentially it's going to end up completely rewriting the `serviceHost.ts` file where we generate the `LanguageServiceHost` to use webpack's input filesystem instead of `ts.sys` for reading files which uses the underlying `fs`.

More coming, but wanted to get some opinions of work as I complete various steps. I've tried to include a little example here: https://github.com/LukeSheard/ts-loader/tree/master/examples/memory-fs

### CODE REVIEW

**Comment by johnnyreilly** [comment]
I'm so glad you're looking into this, thank you! 🌹 

I'm snowed at work right now - but I'll try and take a look at where you've got to this weekend.  Ping me if there's been no action for a while!

File: examples/memory-fs/README.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# vanilla ts-loader
+
+```shell
+yarn install
+
+# Run in watch mode
+yarn watch
```
Comment (johnnyreilly) [inline examples/memory-fs/README.md]: Doesn't look like there actually is a watch task?

**Comment by johnnyreilly** [comment]
Also - it looks like the changes only affect the `config.ts` lookup is that sufficient?

**Comment by opl-** [comment]
@LukeSheard Why was this PR closed?

**Comment by LukeSheard** [comment]
@opl- I just didn't have time to work on it anymore. I agree though - if it's possible it totally should.

**Comment by namesweretaken** [comment]
@LukeSheard What's left to do?

**Pull request closed without merging**